src/main/java/Duke.java

```diff
@@ -1,10 +1,141 @@
+import java.util.Scanner;
+import java.util.Vector;
+
 public class Duke {
+    public class Task {
+        protected String description;
+        protected boolean isDone;
+        protected String taskType;
+        protected String taskDate;
+    
+        public Task(String description,boolean isDone, String taskType, String taskDate) { // initiating an instance
+            this.description = description;
+            this.isDone = isDone;
+            this.taskType = taskType; // todo,ddl,event
```

Comment:
Should use inheritance for task types instead of a string variable since each type has methods and variables unique to them. Also to improve flexibility in future development